### ISSUE TITLE
Bug on find_active_cell_around_point for p::d::t.

### ISSUE THREAD

**Issue by luca-heltai** [dealii/dealii#10175]
PR #10174 reminded us that there is a problem in the fast algorithm, so we had to resort to the slow function
```
          return find_active_cell_around_point(mapping,
                                               mesh,
                                               p,
                                               marked_vertices);
```
in case the initial search did not succeed. This should be investigated and fixed.

**Comment by kronbichler** [comment]
Adjusting to the 9.3 milestone now that we have the temporary solution.

**Comment by luca-heltai** [comment]
So, this has been around for a long time: #7576. I think it is time we think of a permanent solution... (see also #10459, and #10506)

**Comment by nfehn** [comment]
> So, this has been around for a long time: #7576. I think it is time we think of a permanent solution... (see also #10459, and #10506)

@luca-heltai what is the relation to #7930, which you did not mention above?

**Comment by nfehn** [comment]
The problem mentioned above will be solved by PR #10540

**Comment by bangerth** [comment]
@nfehn You mention that this should have been fixed by a patch a couple of years ago. Can you confirm that this is so?

**Comment by kronbichler** [comment]
I think we made several improvements during the last release. However, in https://github.com/exadg/exadg/issues/438 we observed another problematic case. I am not sure if we can fix this entirely for the upcoming 9.5 release, but I think we also need to look over the interfaces, because there are too many functions doing similar things in `GridTools`. Let us keep this on the milestone for now.

**Comment by kronbichler** [comment]
I'm closing this issue, given the progress in #15233. We might have additional bugs or optimizations, but we should make those based on up-to-date test cases and/or code restructuring.
